template/template.go

```diff
@@ -154,6 +155,10 @@ func (t *Template) GetNewContext(folder string, useCache bool) *Template {
 	newTemplate.addFunctions(t.aliases)
 	newTemplate.importTemplates(t)
 	newTemplate.options = make(OptionsSet)
+	if dict := t.Context(); dict.Len() > 0 {
```

Comment:
Even if it is empty, shouldn't it be cloned?

Reply:
If it is empty, this is because the original context is not a dictionary. This could only happen when using gotemplate as a library and in that specific case, the dictionary is freshly created by `t.Context()`. So the answer is no, but you don't miss anything 😄.